packages/app/src/components/DBEditTimeChartForm.tsx

```diff
@@ -525,8 +536,17 @@ export default function EditTimeChartForm({
     [queriedConfig, tableSource, dateRange, queryReady],
   );
 
+  const parentRef = useRef<HTMLElement | null>(null);
+  // Need to force a rerender as the modal will not be mounted when initially rendered
+  const [, forceRerender] = useState(false);
+
+  const setRef = useCallback((el: HTMLElement | null) => {
+    parentRef.current = el;
+    forceRerender(v => !v);
+  }, []);
+
```

Comment:
Instead of using this `setRef` pattern, what if you just used `useState` and get the re-rendering for free?

```
 const [parentElement, setParentElement] = useState<HTMLElement | null>(null);

etc...

Reply:
Good idea 👍 Let me make this change!

Reply:
Updated!